src/chess/chess_game.rs

```diff
@@ -24,18 +26,88 @@ impl ChessMove {
     pub fn new(m: chess::ChessMove) -> Self {
         Self { m: m }
     }
+
     pub fn from_str(pos: &ChessPosition, move_text: &str) -> Result<Self, String> {
         return match chess::ChessMove::from_san(&pos.board, move_text) {
             Ok(m) => Ok(Self::new(m)),
             Err(e) => Err(err_to_str(e)),
         };
     }
+
+    pub fn get_raw(&self) -> &chess::ChessMove {
+        &self.m
+    }
+
+    pub fn to_idx(&self) -> usize {

```

Comment:
this is sort of a quick hash right? what do you think about calling it `quick_hash`? there are too many things called index - this index, the nn index, and `to_idx` in the ttt/hex meaning of "place on the board"

Reply:
or maybe `serial_number`

Reply:
Hash is not what we need, we need a fixed indexing, which doesn't span over a huge range, as MOVE_TO_NN_INDEX size is defined by the max move index.
We can rename some things, maybe call this to_int

Reply:
now i see how it's similar to the ttt/hex definition of `idx`